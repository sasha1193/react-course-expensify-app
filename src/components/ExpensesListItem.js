import React from 'react';
import moment from 'moment';
import { NavLink } from 'react-router-dom';


const ExpensesListItem = ({ id, description, amount, createAt }) => (
    <div>
        <NavLink to={`/edit/${id}`}><h3>{description}</h3></NavLink>
        <p>{amount} - {moment(createAt).format('MMM Do YYYY')}</p>
    </div>
);



export default ExpensesListItem;