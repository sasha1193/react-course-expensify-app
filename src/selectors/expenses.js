//Get visible expenses
import moment from "moment";


export default (expenses, { text, sortBy, startDate, endDate }) => {
    return expenses.filter((expense) => {
        const momentCreatedAt = moment(expense.createAt)
        const startDateMatch = startDate ? startDate.isSameOrBefore(momentCreatedAt, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(momentCreatedAt, 'day') : true;
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());

        return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b) => {
        if(sortBy === 'date') {
            return a.createAt < b.createAt ? 1 : -1;
    } else if (sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
    }
    })
};

