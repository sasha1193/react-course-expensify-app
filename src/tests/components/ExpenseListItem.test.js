import React from 'react';
import { shallow } from 'enzyme';
import ExpensesListItem from '../../components/ExpensesListItem';
import expenses from '../fixtures/expenses';


test('should render ExpenseListItem with expenses', () => {
    const wrapper = shallow(<ExpensesListItem {...expenses[0]} />)
    expect(wrapper).toMatchSnapshot();
})