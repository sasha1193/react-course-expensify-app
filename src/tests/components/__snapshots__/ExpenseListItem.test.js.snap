// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with expenses 1`] = `
<div>
  <NavLink
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </NavLink>
  <p>
    195
     - 
    Jan 1st 1970
  </p>
</div>
`;
